pkg/devfile/adapters/kubernetes/component/kaniko_build.go

```diff
@@ -131,10 +138,11 @@ func (a Adapter) createDockerCfgSecretForInternalRegistry(ns string) error {
 	// internal registry credentials.
 	secret, err := a.getServiceAccountSecret(ns, builderServiceAccount, corev1.SecretTypeDockercfg)
 	if err != nil {
-		return errors.Wrap(err, "failed to get docker secret")
+		return errors.Wrap(err, "failed to retrieve service account credentials")
 	}
-	if err := a.createDockerConfigSecretFrom(secret); err != nil {
-		return errors.Wrap(err, "failed to create docker secret")
+
+	if err := a.createDockerConfigSecretFrom(secret, secret.GetNamespace()); err != nil {
```

Comment:
Don't pass `namespace` as a seperate argument.   We can get it from the `secret` argument.  Don't make it argument unncessarly long.

Reply:
fixed